integration-tests/src/env/setup.rs

```diff
@@ -460,7 +461,7 @@ pub fn setup_client_with_runtime(
     config.epoch_length = chain_genesis.epoch_length;
     let protocol_upgrade_schedule = get_protocol_upgrade_schedule(&chain_genesis.chain_id);
     let multi_spawner = AsyncComputationMultiSpawner::default()
-        .custom_apply_chunks(Arc::new(RayonAsyncComputationSpawner)); // Use rayon instead of the default thread pool 
+        .custom_apply_chunks(Arc::new(StdThreadAsyncComputationSpawnerForTest)); // Use std::thread instead of the default thread pool 
```

Comment:
Changed this because with rayon sometimes one task (eg, receiver) could get scheduled and others (workers) may not. With many tests running together some tests timeout for me.

I guess we could put this back by changing the worker/receiver pattern such that each worker is identical and pulls the jobs from a channel, then if there are no jobs left the worker would wait for the other work to complete and then do the callback, but this seems more complicated to me.

Also open to other ideas.

Reply:
Actually ended putting this part back, as the above is not correct and each test runs its own process (thanks to @nagisa for mentioning it to me). Yet rayon tasks seemingly can get stuck in this fashion (possibly both scheduled to same queue and work stealing not happening?), with logs I confirmed both tasks are spawned but the sender does not start.

Instead I changed it to put results in corresponding indexes, using locks for cross-thread mutability.
It is not pretty but it works with Rayon or StdThread backing.